__tests__/src/components/TicketsTable.test.tsx

```diff
@@ -137,18 +137,19 @@ describe('TicketsTable Sorting', () => {
     expect(categoryRows[2]).toHaveTextContent('pet_fostering');
   });
 
-  it('Testing Date or created_at Sort', () => {
-    render(<TicketsTable items={items} />);
+  it('sorts tickets from oldest to newest', () => {
+    render(<TicketsTable items={mockServiceRequestSummaries} />);
     const dateHeader = screen.getByText('Date');
     fireEvent.click(dateHeader);
 
     const rows = screen.getAllByRole('row');
-    expect(rows[1]).toHaveTextContent('09/09/2024');
-    expect(rows[2]).toHaveTextContent('09/09/2024');
-    expect(rows[3]).toHaveTextContent('09/10/2024');
+    const dates = rows.slice(3).map((row) => (row as HTMLTableRowElement).cells[4].textContent); // assuming the date is the 4th cell    dates.push('9/9/2010'); // This will need to be removed, I was just making sure it broke correctly
+
+    const sortedDates = [...dates].sort((a, b) => new Date(a).getTime() - new Date(b).getTime());
```

Comment:
The problem with this is that if all the dates are the same, then the test won't tell you anything. When I was testing the code locally this is what I found. I think what needs to happen is that the test data code should not contain random dates

Reply:
Like if you comment out remove the sorting [here](https://github.com/digitalaidseattle/seattle-humane-app/pull/94/files#diff-4625abbf15ef6428ed1aea3927a09200dbe94723485aca7c1952db8784b6ed00R62-R66), you'll see the test still passes, which is not great. I was poking around to see if I could fix it. By changing the seed [here](https://github.com/digitalaidseattle/seattle-humane-app/blob/4acb5ea989c2f9a3121661eac57ae57948a251a4/utils/TestData.ts#L9) and adding some kind of date generator to the `TestData.ts` file I had some success, but it doesn't feel like the right solution completely.

``` TypeScript
// Create a reference date
const REFERENCE_DATE = new Date('2024-01-01T00:00:00.000Z');

// Helper function to generate consistent dates
const generateConsistentDate = (daysToSubtract: number) => {
  const days = Math.max(1, daysToSubtract); // Ensure days is at least 1
  return date.recent({ days, refDate: REFERENCE_DATE });
};
```

and

```TypeScript
const genMockTicket: (idx: number) => ServiceRequestType = (idx: number) => ({
  id: string.uuid(),
  client_id: mockClients[idx % mockClients.length].id,
  pet_id: mockPets[idx % mockPets.length].id,
  service_category: MockAppConstants.category[0].value,
  request_source: MockAppConstants.source[0].value,
  status: MockAppConstants.status[idx % MockAppConstants.status.length].id,
  team_member_id: mockTeamMembers[idx % mockTeamMembers.length].id,
  description: lorem.sentence(),
  created_at: generateConsistentDate(5 - idx).toISOString(),
  urgent: boolean({ probability: 0.5 }),
  modified_at: generateConsistentDate(4 - idx).toISOString(), // Ensure modified_at is after created_at
});
```

Reply:
As discussed on slack, maybe it's best to just not use testdata.ts

Reply:
To further elaborate - Tristan and I had a discussion on a private chat where we agreed that using TestData.ts overcomplicates the PR. The test data that is generated is a ton of random tickets, whereas Tristan needs specific test data.

Reply:
I think it's ok - faker is useful sometimes but it is a little messier than test data in the code. I certainly don't have perfect to be the enemy of good. It's most important that we have tests that will protect our code against regressions